File: Julia/lambdaconscarcdr.jl
#!/usr/bin/env julia

kons  = (hd, tl = nothing) -> x -> if x; hd else tl end
virst = l -> l(true )
rrest = l -> l(false)

lstt  = kons(1, kons(2))

virst(rrest(lstt))

File: Elixir/lambdaconscarcdr.exs
#!/usr/bin/env elixir

kons  = fn hd, tl -> &(if &1 do hd else tl end) end
virst = &(&1.(true ))
rrest = &(&1.(false))

lstt  = kons.(1, kons.(2, nil))

virst.(rrest.(lstt))
